.changeset/brown-tips-drop.md

```diff
@@ -0,0 +1,5 @@
+---
+'astro': minor
+---
+
+Plugins can now access a page's boolean exports on RouteData['customOptions']
```

Comment:
I would call it `externalOptions`. It's more sound and communicates that these options **don't** from Astro. Or `pluginOptions`.

Reply:
I don't think we should differentiate. If we accept this feature I think it should include all boolean options, including `prerender`.

The reason is, if this field is just for non-Astro defined booleans, that would mean that if Astro adds any new booleans itself, that becomes a breaking change. However if they are all under `customOptions` then there is no breaking change.

Reply:
Another option here is that custom options must have some prefix we define. Similar to how HTML has the `data-` attributes. If they were required to be `customFoo`, for example, we could ensure that Astro never accidentally breaks an integration defined option.

Reply:
I like that idea but `custom` as a prefix feels a bit bulky to me